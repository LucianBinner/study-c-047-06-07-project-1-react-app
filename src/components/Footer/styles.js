import styled from 'styled-components';
import { theme } from '../../styles/theme';

export const Container = styled.div`
  text-align: center;

  a {
    color: inherit;
    text-decoration: none;
    font-size: ${theme.font.sizes.small};
  }
`;
