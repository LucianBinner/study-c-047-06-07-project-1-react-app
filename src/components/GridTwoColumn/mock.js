export default {
  title: 'Grid Two Column',
  text: `Lorem ipsum, dolor sit amet consectetur adipisicing elit. Temporibus, culpa impedit dolore assumenda voluptatibus aperiam quae quasi minus dolor eum illum voluptate minima nisi corporis nesciunt at, molestias pariatur quaerat?`,
  srcImg: 'assets/images/javascript.svg',
  background: true,
};
