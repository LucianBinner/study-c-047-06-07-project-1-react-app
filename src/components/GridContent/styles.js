import styled from 'styled-components';
import { theme } from '../../styles/theme';

export const Container = styled.div`
  text-align: center;
  max-width: 55rem;
  margin: 0 auto;
`;

export const Html = styled.div`
  margin: ${theme.spacings.xhuge} 0;
`;
