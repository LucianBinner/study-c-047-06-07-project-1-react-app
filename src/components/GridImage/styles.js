import styled from 'styled-components';
import { theme } from '../../styles/theme';
import { Container as TextComponent } from '../TextComponent/styles';

export const Container = styled.div`
  > ${TextComponent} {
      margin-bottom: ${theme.spacings.xhuge};
    }
`;

export const Grid = styled.div`
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
  gap: ${theme.spacings.large};
`;

export const GridElement = styled.div`
  overflow: hidden;
`;

export const Image = styled.img`
  width: 100%;
  transition: all 300ms ease-in-out;

  &:hover {
    transform: scale(1.2) rotate(10deg);
  }
`;
