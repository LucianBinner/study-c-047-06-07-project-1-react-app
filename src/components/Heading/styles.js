import styled, { css } from 'styled-components';
import { theme } from '../../styles/theme';

const titleSize = {
  small: () => css`
    font-size: ${theme.font.sizes.medium};
  `,
  medium: () => css`
    font-size: ${theme.font.sizes.large};
  `,
  big: () => css`
    font-size: ${theme.font.sizes.xlarge};
  `,
  huge: () => css`
    font-size: ${theme.font.sizes.xhuge};
    ${mediaFont(theme)};
  `,
};

const mediaFont = () => css`
  @media ${theme.media.lteMedium} {
    font-size: ${theme.font.sizes.xlarge};
  }
`;

const titleCase = (uppercase) => css`
  text-transform: ${uppercase ? 'uppercase' : 'none'};
`;

export const Title = styled.h1`
  color: ${(props) =>
    props.colorDark ? theme.colors.primaryColor : theme.colors.white};

  ${({ size }) => titleSize[size]()}
  ${({ uppercase }) => titleCase(uppercase)}
`;
