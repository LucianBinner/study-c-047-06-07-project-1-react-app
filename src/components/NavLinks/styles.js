import styled from 'styled-components';
import { theme } from '../../styles/theme';

export const Container = styled.div`
  display: flex;
  flex-flow: row wrap;

  @media ${theme.media.lteMedium} {
    flex-flow: column wrap;
    align-content: center;
  }
`;
