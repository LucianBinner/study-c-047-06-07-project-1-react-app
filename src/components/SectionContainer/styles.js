import styled from 'styled-components';
import { theme } from '../../styles/theme';

export const Container = styled.div`
  max-width: 120rem;
  margin: 0 auto;
  padding: ${theme.spacings.large};
`;
