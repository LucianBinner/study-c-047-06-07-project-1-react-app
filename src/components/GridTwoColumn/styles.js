import styled from 'styled-components';
import { theme } from '../../styles/theme';
import { Title } from '../Heading/styles';

export const Container = styled.div`
  display: grid;
  grid-template-columns: 1fr 2fr;
  align-items: center;
  gap: ${theme.spacings.large};

  @media ${theme.media.lteMedium} {
    grid-template-columns: 1fr;
    text-align: center;
  }

  ${Title} {
    margin-bottom: ${theme.spacings.large};
  }
`;

export const TextContainer = styled.div`
  @media ${theme.media.lteMedium} {
    margin-bottom: ${theme.spacings.large};
  }
`;

export const ImageContainer = styled.div``;

export const Image = styled.img``;
