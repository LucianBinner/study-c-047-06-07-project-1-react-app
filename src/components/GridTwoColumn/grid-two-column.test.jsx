import { renderTheme } from '../../styles/render-theme';
import { GridTwoColumn } from '.';
import mock from './mock';

describe('<GridTwoColumn />', () => {
  it('Should render two column grid', () => {
    const { container } = renderTheme(<GridTwoColumn {...mock} />);
    expect(container).toMatchSnapshot();
  });

  it('Should render two column grid with background undefined', () => {
    const { container } = renderTheme(
      <GridTwoColumn {...mock} background={undefined} />,
    );
    expect(container).toMatchSnapshot();
  });
});
