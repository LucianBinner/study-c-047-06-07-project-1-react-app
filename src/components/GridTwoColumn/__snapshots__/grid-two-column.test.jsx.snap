// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridTwoColumn /> Should render two column grid 1`] = `
.c5 {
  color: #ffffff;
  font-size: 4.0rem;
  text-transform: uppercase;
}

.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
}

.c0 {
  background: #ffffff;
  color: #0a1128;
  background: #0a1128;
  color: #ffffff;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c6 {
  font-size: 2.4rem;
  font-weight: 400;
}

.c2 {
  display: grid;
  grid-template-columns: 1fr 2fr;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 3.2rem;
}

.c2 .c4 {
  margin-bottom: 3.2rem;
}

@media (max-width:768px) {
  .c2 {
    grid-template-columns: 1fr;
    text-align: center;
  }
}

@media (max-width:768px) {
  .c3 {
    margin-bottom: 3.2rem;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          <h2
            class="c4 c5"
          >
            Grid Two Column
          </h2>
          <div
            class="c6"
          >
            Lorem ipsum, dolor sit amet consectetur adipisicing elit. Temporibus, culpa impedit dolore assumenda voluptatibus aperiam quae quasi minus dolor eum illum voluptate minima nisi corporis nesciunt at, molestias pariatur quaerat?
          </div>
        </div>
        <div
          class=""
        >
          <img
            alt="Grid Two Column"
            class=""
            src="assets/images/javascript.svg"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<GridTwoColumn /> Should render two column grid with background undefined 1`] = `
.c5 {
  color: #0a1128;
  font-size: 4.0rem;
  text-transform: uppercase;
}

.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
}

.c0 {
  background: #ffffff;
  color: #0a1128;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c6 {
  font-size: 2.4rem;
  font-weight: 400;
}

.c2 {
  display: grid;
  grid-template-columns: 1fr 2fr;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 3.2rem;
}

.c2 .c4 {
  margin-bottom: 3.2rem;
}

@media (max-width:768px) {
  .c2 {
    grid-template-columns: 1fr;
    text-align: center;
  }
}

@media (max-width:768px) {
  .c3 {
    margin-bottom: 3.2rem;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          <h2
            class="c4 c5"
          >
            Grid Two Column
          </h2>
          <div
            class="c6"
          >
            Lorem ipsum, dolor sit amet consectetur adipisicing elit. Temporibus, culpa impedit dolore assumenda voluptatibus aperiam quae quasi minus dolor eum illum voluptate minima nisi corporis nesciunt at, molestias pariatur quaerat?
          </div>
        </div>
        <div
          class=""
        >
          <img
            alt="Grid Two Column"
            class=""
            src="assets/images/javascript.svg"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
